src/hotspot/cpu/aarch64/icBuffer_aarch64.cpp

```diff
@@ -52,9 +52,16 @@ void InlineCacheBuffer::assemble_ic_buffer_code(address code_begin, void* cached
   address start = __ pc();
   Label l;
   __ ldr(rscratch2, l);
-  __ far_jump(ExternalAddress(entry_point));
-  __ align(wordSize);
+  int jump_code_size = __ far_jump(ExternalAddress(entry_point));
+  // IC stub code size is not expected to vary depending on target address.
+  // We use NOPs to make the ldr+far_jump+int64 size equal to ic_stub_code_size.
+  for (int i = jump_code_size;
+           i < ic_stub_code_size() - 3 * NativeInstruction::instruction_size;
+           i += NativeInstruction::instruction_size) {
+    __ nop();
+  }
   __ bind(l);
+  assert((uintptr_t)__ pc() % wordSize == 0);
```

Comment:
Looks like this leads to a compilation failure in the debug build. Can you please check before submitting?